### ISSUE TITLE
Repurposing doe/aes

### ISSUE THREAD

**Issue by RJMJJJ** [chipsalliance/caliptra-rtl#416]
Can we repurpose the existing doe/aes RTL modules, by adding them as new slave on AHB-L to perform encryption/decryption on f/w fetched from Flash. We are exploring active mode implementation and hence the Flash (QSPI) is accessed by RISC-V (in Caliptra). 
What I understand these doe/aes RTL modules do not have side channel mitigation capability. But, apart from this which of the doe/aes is better suitable candidate for mentioned purpose.
The doc covers the kind of AES support by doe but there is no mention of what capability aes module have. Why there are two different modules with different set of files?
Kindly address this - as we don't have any other AES IP available, we would like to experiment with this. So, which one of doe/aes module to pick up is important decision for us.

**Comment by bharatpillilli** [comment]
- First gen Caliptra doesn't have encryption/decryption usage models. 
- Any code can be reused as long as it is understood its usage mode (eg. trying to use AES for ency/decry without masking is not recommended by security expectation 
- While any code can be repurposed/enhanced, we cant be held responsible for it since its not Caliptra trademarked. 

Since I got those legal type of statements out of the way, as I communicated yesterday, Caliptra does not have AES. We reused a version of AES for DOE purposes and it meets our requirements (security, side channel etc.). 

Future Caliptra version may include an AES engine but its not yet decided.
